karabo_data/detector_data.py

```diff
@@ -0,0 +1,393 @@
+import numpy as np
+import pandas as pd
+import xarray
+
+from .reader import DataCollection, by_id, by_index
+
+MAX_PULSES = 2700
```

Comment:
This is used when you leave a slice open at the end, like `[5:]`, so it shouldn't matter if it's greater than the maximum. And it can be used to select pulses by ID as well, which might go beyond the memory cell depth.